src/main/java/com/google/sps/servlets/Search.java

```diff
@@ -0,0 +1,97 @@
+import com.google.sps.util.SearchUtils;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Handles requests for a Datastore search based on a query string with filters to use.
+ **/
+@WebServlet("/search")
+public class Search extends HttpServlet {
+  /**
+   * Hnadles the get request for Datastore search results. Processes the url search parameters
+   * and returns a list in JSON format with all the recipe information to be rendered client side.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Initializes a user query object which will seperate the query data for ease of access.
+    UserQuery uQuery = new UserQuery(request.getParameter("query"));
+    
+    // Creates the new Query and adds the filter to be used in Datastore
+    // The filter is currently only set to find an exact match name in Datastore
+    // TODO: Composite Query Filter will be needed to combine and add all filter to query in future
+    Query query = new Query("PublicRecipe");
+    query.setFilter(SearchUtils.getRecipeNameFilter(uQuery.getName()));
+
+    // Makes the query to the datastore and converts it to a list so it can operated on.
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery datastoreResult = datastore.prepare(query);
+    List<Entity> recipes = datastoreResult.asList(FetchOptions.Builder.withLimit(10));
+
+    // Create a list of recipe entity information to be sent to the client, and then iterate through
+    // the datastore response to create the recipe objects and add them to the list
+    ArrayList<Recipe> clientRecipeInfo = new ArrayList<Recipe>();
+    for (int i = 0; i < recipes.size(); i++) {
+      Entity datastoreRecipe = recipes.get(i);
+      clientRecipeInfo.add(
+        new Recipe(
+          datastoreRecipe.getKey().getId(),
+          (String) datastoreRecipe.getProperty("name"),
+          (String) datastoreRecipe.getProperty("imageBlobKey"),
+          (String) datastoreRecipe.getProperty("description")
+        )
+      );
+    }
+
+    // Sets the response type,converts the data into JSON, and sends the response
+    response.setContentType("JSON");
+    String recipeResponse = convertToJson(clientRecipeInfo);
+    response.getWriter().println(recipeResponse);
+  }
+
+  /**
+   * Simple Implementation to reduce need to type print statements in debugging.
+   **/
+  private void SOP(Object thing) {
+    System.out.println(thing);
+  }
+
+  /**
+   * Converts an object into JSON using Gson, but abstracts the need to make
+   * a Gson object in different parts of code.
+   **/
+  private String convertToJson(Object target) {
+    Gson gson = new Gson();
+    return gson.toJson(target);
+  }
```

Comment:
These seem like pretty generic methods. If they'd be useful for other classes as well, consider moving them to a utils file

Reply:
This is a good point. I've been using them many other places, so this would eliminate the need to copy/paste